compile errors

Hi,

I had to change some makefiles to get pokerstove compiled: 
added -std=c++11 to CMAKE_CXX_FLAGS in ./src/lib/pokerstove/penum/CMakeLists.txt

and -lpthread to ./CMakeFiles/util_tests.dir/link.txt

In addition to what @thomaxxl mentioned I needed to put that -std=c++11 also in CMakeLists.txt files in peval and util as well.

So to sum it up, the following changes are neccessary to get the newest commit (0f057ff5168d18754dfda154f4715281f8a678ee) to compile:

Add -std=c++11 in CMAKE_CXX_FLAGS in following files:
- ./src/lib/pokerstove/penum/CMakeLists.txt
- ./src/lib/pokerstove/penum/CMakeLists.txt
- ./src/lib/pokerstove/util/CMakeLists.txt

After that run cmake and make. It will stop on error when trying to build the testkit (undefined reference to some pthread stuff). So at that point you need to modify the compiled file in {build_dir}/lib/pokerstove/util/CMakeFiles/util_tests.dir/link.txt and add -lpthread in the end. Now run make again and it should compile.

I could add a push request but I'm not sure how to address the problem with googletest properly.

--- a/.travis.yml
+++ b/.travis.yml
@@ -19,6 +19,6 @@ before_script:
 
 script:
     - cd build
-    - cmake -DCMAKE_CXX_FLAGS='-std=c++11' -DCMAKE_VERBOSE_MAKEFILE=ON ..
+    - cmake -DCMAKE_VERBOSE_MAKEFILE=ON ..
     - make -j 2
     - make test

--- a/.travis.yml
+++ b/.travis.yml
@@ -19,6 +19,6 @@ before_script:
 
 script:
     - cd build
-    - cmake -DCMAKE_CXX_FLAGS='-std=c++11' -DCMAKE_VERBOSE_MAKEFILE=ON ../src
+    - cmake -DCMAKE_CXX_FLAGS='-std=c++11' -DCMAKE_VERBOSE_MAKEFILE=ON ..
     - make -j 2
     - make test

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -9,6 +9,9 @@ project(pokerstove)
 #       but since all of them require C++11, the flag is set globally.
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++0x")
 
+# TODO: Remove the depricated register keywords
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-deprecated-register")
+
 # Set up gtest. This must be set up before any subdirectories are
 # added which will use gtest.
 add_subdirectory(src/ext/googletest)

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,0 +1,42 @@
+cmake_minimum_required(VERSION 2.6)
+
+project(pokerstove)
+
+# Set up gtest. This must be set up before any subdirectories are
+# added which will use gtest.
+add_subdirectory(src/ext/googletest)
+find_library(gtest REQUIRED)
+include_directories(${GTEST_INCLUDE_DIRS})
+link_directories(${GTEST_LIBS_DIR})
+add_definitions("-fPIC")
+enable_testing()
+
+#
+# set up boost
+#
+# The goal is to make this work for linux/win/osx
+if (UNIX)
+    message(status "building for unix")
+    set(BOOST_LIBRARYDIR "/usr/local/lib/")
+elseif(CMAKE_GENERATOR STREQUAL Xcode)
+    message(status ": building for xcode")
+    set(BOOST_ROOT "/opt/include/")
+    set(BOOST_LIBRARYDIR "/opt/local/lib")
+else()
+    message(status ": building for non-unix -> windows")
+    set(BOOST_ROOT "E:/code/ext/boost_1_53_0/")
+    set(BOOST_LIBRARYDIR "E:/code/ext/boost_1_53_0/lib32-msvc-10.0/")
+endif()
+set(Boost_USE_STATIC_LIBS   OFF)
+set(Boost_USE_MULTITHREADED      ON)
+find_package(Boost COMPONENTS program_options REQUIRED)
+include_directories(${Boost_INCLUDE_DIR})
+link_directories(${Boost_LIBRARY_DIRS})
+message(status "** Boost Include: ${Boost_INCLUDE_DIR}")
+message(status "** Boost Libraries: ${Boost_LIBRARY_DIRS}")
+message(status "** Boost Libraries: ${Boost_LIBRARIES}")
+
+# Also search for includes in PROJECT_BINARY_DIR to find config.h.
+include_directories("${PROJECT_BINARY_DIR}")
+set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
+add_subdirectory(src)

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -24,21 +24,8 @@ enable_testing()
 #
 # set up boost
 #
-# The goal is to make this work for linux/win/osx
-if (UNIX)
-    message(status "building for unix")
-    set(BOOST_LIBRARYDIR "/usr/local/lib/")
-elseif(CMAKE_GENERATOR STREQUAL Xcode)
-    message(status ": building for xcode")
-    set(BOOST_ROOT "/opt/include/")
-    set(BOOST_LIBRARYDIR "/opt/local/lib")
-else()
-    message(status ": building for non-unix -> windows")
-    set(BOOST_ROOT "E:/code/ext/boost_1_53_0/")
-    set(BOOST_LIBRARYDIR "E:/code/ext/boost_1_53_0/lib32-msvc-10.0/")
-endif()
-set(Boost_USE_STATIC_LIBS   OFF)
-set(Boost_USE_MULTITHREADED      ON)
+set(Boost_USE_STATIC_LIBS OFF)
+set(Boost_USE_MULTITHREADED ON)
 find_package(Boost COMPONENTS program_options REQUIRED)
 include_directories(${Boost_INCLUDE_DIR})
 link_directories(${Boost_LIBRARY_DIRS})

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2,6 +2,13 @@ cmake_minimum_required(VERSION 2.6)
 
 project(pokerstove)
 
+# TODO: Discover C++11 support.
+# TODO: This flags only works for GCC compatible compilers (Clang, Intel).
+#
+# TODO: This, ideally, should be per library requirement,
+#       but since all of them require C++11, the flag is set globally.
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++0x")
+
 # Set up gtest. This must be set up before any subdirectories are
 # added which will use gtest.
 add_subdirectory(src/ext/googletest)

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -29,9 +29,9 @@ set(Boost_USE_MULTITHREADED ON)
 find_package(Boost COMPONENTS program_options REQUIRED)
 include_directories(${Boost_INCLUDE_DIR})
 link_directories(${Boost_LIBRARY_DIRS})
-message(status "** Boost Include: ${Boost_INCLUDE_DIR}")
-message(status "** Boost Libraries: ${Boost_LIBRARY_DIRS}")
-message(status "** Boost Libraries: ${Boost_LIBRARIES}")
+message(STATUS "Boost Include: ${Boost_INCLUDE_DIR}")
+message(STATUS "Boost Libraries: ${Boost_LIBRARY_DIRS}")
+message(STATUS "Boost Libraries: ${Boost_LIBRARIES}")
 
 # Also search for includes in PROJECT_BINARY_DIR to find config.h.
 include_directories("${PROJECT_BINARY_DIR}")

--- a/README.md
+++ b/README.md
@@ -56,8 +56,8 @@ To build under linux using cmake, create a build directory,
 invoke cmake on the programs directory, then build.
 
     git clone https://github.com/andrewprock/pokerstove.git
-    mkdir pokerstove/src/build
-    cd pokerstove/src/build
+    mkdir pokerstove/build
+    cd pokerstove/build
     cmake ..
     make
 

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -1,53 +1,12 @@
-cmake_minimum_required(VERSION 2.6)
-
-project(pokerstove)
-
-#
-# set up boost
-#
-# The goal is to make this work for linux/win/osx
-if (UNIX)
-    message(status "building for unix")
-    set(BOOST_LIBRARYDIR "/usr/local/lib/")
-elseif(CMAKE_GENERATOR STREQUAL Xcode)
-    message(status ": building for xcode")
-    set(BOOST_ROOT "/opt/include/")
-    set(BOOST_LIBRARYDIR "/opt/local/lib")
-else()
-    message(status ": building for non-unix -> windows")
-    set(BOOST_ROOT "E:/code/ext/boost_1_53_0/")
-    set(BOOST_LIBRARYDIR "E:/code/ext/boost_1_53_0/lib32-msvc-10.0/")
-endif()
-set(Boost_USE_STATIC_LIBS   OFF)
-set(Boost_USE_MULTITHREADED      ON)
-find_package(Boost COMPONENTS program_options REQUIRED)
-include_directories(${Boost_INCLUDE_DIR})
-link_directories(${Boost_LIBRARY_DIRS})
-message(status "** Boost Include: ${Boost_INCLUDE_DIR}")
-message(status "** Boost Libraries: ${Boost_LIBRARY_DIRS}")
-message(status "** Boost Libraries: ${Boost_LIBRARIES}")
-
-# Set up gtest. This must be set up before any subdirectories are
-# added which will use gtest.
-add_subdirectory(ext/googletest)
-find_library(gtest REQUIRED)
-include_directories(${GTEST_INCLUDE_DIRS})
-link_directories(${GTEST_LIBS_DIR})
-add_definitions ("-fPIC")
-enable_testing()
-
 # check_function_exists does not work for builtins, so use try_compile.
 try_compile(HAVE_BUILTIN_BITOPS
             "${PROJECT_BINARY_DIR}"
-            "${PROJECT_SOURCE_DIR}/TestBuiltinBitops.cpp")
+            "${CMAKE_CURRENT_SOURCE_DIR}/TestBuiltinBitops.cpp")
 
 # Set up the configure file.
-configure_file("${PROJECT_SOURCE_DIR}/Config.h.in"
-               "${PROJECT_BINARY_DIR}/Config.h")
+configure_file(Config.h.in Config.h)
 
-# Also search for includes in PROJECT_BINARY_DIR to find config.h.
-include_directories(lib/ "${PROJECT_BINARY_DIR}")
-set (CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
+include_directories(lib/)
 add_subdirectory(lib/pokerstove/peval)
 add_subdirectory(lib/pokerstove/penum)
 add_subdirectory(lib/pokerstove/util)

--- a/src/lib/pokerstove/penum/CMakeLists.txt
+++ b/src/lib/pokerstove/penum/CMakeLists.txt
@@ -1,8 +1,6 @@
 cmake_minimum_required(VERSION 2.8)
 find_package (Threads)
 
-set (CMAKE_CXX_FLAGS "-O2")
-
 # build the test
 file(GLOB test_sources *.test.cpp)
 add_executable(penum_tests ${test_sources})

--- a/src/lib/pokerstove/util/CMakeLists.txt
+++ b/src/lib/pokerstove/util/CMakeLists.txt
@@ -1,8 +1,6 @@
 cmake_minimum_required(VERSION 2.8)
 
-set (CMAKE_CXX_FLAGS "-Wno-deprecated-register")
-
-# build the test 
+# build the test
 file(GLOB test_sources *.test.cpp)
 add_executable(util_tests ${test_sources})
 target_link_libraries(util_tests peval gtest gtest_main)

--- a/src/programs/CMakeLists.txt
+++ b/src/programs/CMakeLists.txt
@@ -1,7 +1,3 @@
-cmake_minimum_required(VERSION 2.6)
-
-project(programs)
-
 add_subdirectory (ps-eval)
 add_subdirectory (ps-colex)
 add_subdirectory (ps-lut)

--- a/src/programs/ps-colex/CMakeLists.txt
+++ b/src/programs/ps-colex/CMakeLists.txt
@@ -1,7 +1,6 @@
 project(eval)
 
 add_executable(ps-colex main.cpp)
-add_definitions ("-std=c++0x")
 
 target_link_libraries(ps-colex
         peval

--- a/src/programs/ps-eval/CMakeLists.txt
+++ b/src/programs/ps-eval/CMakeLists.txt
@@ -1,8 +1,6 @@
 project(eval)
 
 add_executable(ps-eval main.cpp)
-#add_definitions ("-ansi -Wall -std=c++0x")
-add_definitions ("-std=c++0x")
 
 target_link_libraries(ps-eval
         peval

--- a/src/programs/ps-lut/CMakeLists.txt
+++ b/src/programs/ps-lut/CMakeLists.txt
@@ -1,7 +1,6 @@
 project(eval)
 
 add_executable(ps-lut main.cpp)
-add_definitions ("-std=c++11")
 
 target_link_libraries(ps-lut
         peval

